### PR TITLE
Improve message body output from plain text editor

### PR DESCRIPTION
**Pull request by alunturner**
When writing a message with mentions in the plain text editor, the `body` content on the sent message was not what we wanted - it was the html from the composer and so contained spurious html. 

This change fixes that and formats a mention in the message.body as expected for a room, user and at room mention.

Before:
Element__2____Alun_Turner

After:
Element__2____Alun_Turner

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Improve message body output from plain text editor ([\#11124](https://github.com/matrix-org/matrix-react-sdk/pull/11124)). Contributed by @alunturner.

### CODE REVIEW

File: test/components/views/rooms/wysiwyg_composer/utils/createMessageContent-test.ts [hunk 1]
```diff
@@ -41,117 +41,146 @@ describe("createMessageContent", () => {
...
+
+        it("Should replace room mentions with room mxid in body", async () => {
+            const messageComposerState = `<a href="https://matrix.to/#/#test_room:element.io" contenteditable="false" data-mention-type="room" style="some styling">a test room</a> `;
+
+            const content = await createMessageContent(messageComposerState, false, { permalinkCreator });
+
+            expect(content).toMatchObject({
+                body: "#test_room:element.io ",
+            });
+        });
```
Comment (alunturner) [inline test/components/views/rooms/wysiwyg_composer/utils/createMessageContent-test.ts:184]: These are the new tests, all other changes in this file result from increased nesting

**Review by t3chguy — APPROVED** [review]
Looks sane

**Comment by t3chguy** [comment]
@alunturner shouldn't this be done in the rust side so its shared between all the consumers so its consistent between Android & iOS too.

**Comment by alunturner** [comment]
In the future, I think that's where we want to end up. Unfortunately it's not a viable approach at the moment.

When the user is entering input into the plain text mode, there is no model present (at all). I'd like the solution to this to be that when a user sends a message from the plain text mode, we take their input, create a rust model from that input, then use that rust model to generate both `body` and `formatted_body` strings. Then we have consistency and a clear source of truth (ie we know message content is always generated by the rust model).

The reason why this is currently not a valid approach is that there are considerable inconsistencies in parsing MD input/output in the model. This means that if we take a MD message, generate a rust model, then get it to output the MD, it will be different to the input message in quite a few cases (can be seen when toggling between rich and plain modes with markdown input in the composer).

**Pull request merged**